deepee/wrapper.py

```diff
@@ -176,19 +176,39 @@ def clip_and_accumulate(self, reduce: str = "mean") -> None:
         for model in self.models:
             torch.nn.utils.clip_grad_norm_(model.parameters(), self.L2_clip)
 
-        model_grads = zip(
-            *[
-                [param.grad for param in m.parameters() if param.requires_grad]
-                for m in self.models
+        clone_grads = [
+            [param.grad for param in m.parameters() if param.requires_grad]
+            for m in self.models
+        ]
+        clone_lens = [len(c) for c in clone_grads]
+        model_grads = zip(*clone_grads)
+
+        try:
+            wrapped_grads = [
+                param
+                for param in self.wrapped_model.parameters()
+                if param.requires_grad
             ]
-        )
-
-        for param, gradient_source in zip(
-            [param for param in self.wrapped_model.parameters() if param.requires_grad],
-            model_grads,
-        ):
-            if param.requires_grad:
-                param.grad = reduction(torch.stack(gradient_source), dim=0)
+            L = len(wrapped_grads)
+            if any([L != c for c in clone_lens]):
+                raise RuntimeError(
+                    "Clone grads and wrapped grads have different size. "
+                    "Did you change the model without calling update_clones()?"
+                )
+            for param, gradient_source in zip(
+                wrapped_grads,
+                model_grads,
+            ):
+                if param.requires_grad:
+                    param.grad = reduction(torch.stack(gradient_source), dim=0)
+        except RuntimeError as e:
+            if "assigned grad has data of a different size" in str(e):
+                raise RuntimeError(
+                    "Grads of wrapped model and clones are not matching. "
+                    "Did you change the model without calling update_clones()?"
+                ) from e
+            else:
+                raise e
```

Comment:
Pedantic comment: In 211, you could get away with just `raise`, which re-raises the last error 😲